tools/bigtable-backup/bigtable-backup.py

```diff
@@ -86,12 +73,62 @@ def ensure_backups(args):
         if table_number != active_table_number and len(timestamps) > 1:
             for timestamp in timestamps[:-1]:
                 delete_backup(table_id, str(timestamp), args)
+                num_backups_deleted += 1
 
+    if args.delete_out_of_range_backups:
+        num_backups_deleted += delete_out_of_range_backups(oldest_table_number, newest_table_number, backups, args)
+
+    set_ensure_backups_specific_metrics(args, num_backups_deleted, newest_table_number, active_table_number)
     push_job_finished_metric(args.prom_push_gateway_endpoint, args.namespace, job_ensure_backups, int(time.time() - start_time))
 
 def find_last_timestamp_from_table_number(table_number, periodic_secs):
     return ((table_number + 1) * periodic_secs) - 1
 
+def list_backups(backup_path):
+    popen = subprocess.Popen(['bigtable-backup', 'list-backups', '-ojson', '--backup-path', backup_path],
+                             stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+    popen.wait()
+
+    return json.loads(popen.stdout.readline())
+
+def set_ensure_backups_specific_metrics(args, num_backups_deleted, newest_table_number, active_table_number):
+    # ensure-backups job specific metrics
+    bigtable_backup_job_tables_backed_up = Gauge('bigtable_backup_job_tables_backed_up', 'Number of backups for active and inactive tables found during last run', ['kind'], registry=registry)
+    bigtable_backup_job_backups = Gauge('bigtable_backup_job_backups', 'Sum of number of backups for all active and inactive tables found during last run', ['kind'], registry=registry)
+    bigtable_backup_job_backups_deleted = Gauge('bigtable_backup_job_backups_deleted', 'Number of backups deleted during last run', registry=registry)
+    bigtable_backup_job_expected_backups = Gauge('bigtable_backup_job_expected_backups', 'Number of backups expected for active and inactive tables', ['kind'], registry=registry)
```

Comment:
I see the issue being that this script _expects_ itself to be run every day. This'll fail for active table if it's being run ever other day for example.

Reply:
What do we get by knowing the _expected_ backups for active table?

Further, I think it's better to expose the _last_ backup time for the active table. That way will be able to an alert like active_backup is less than 36hrs old, etc.

Reply:
right, the script should not know how frequently it is going to run.
removing expected number of backups for active table and adding timestamp of last last backup of active table.